src/clib/pioc.c

```diff
@@ -804,7 +804,10 @@ int PIOc_Init_Intracomm(MPI_Comm comp_comm, int num_iotasks, int stride, int bas
         return check_mpi2(NULL, NULL, mpierr, __FILE__, __LINE__);
 
     /* Check the inputs. */
-    if (!iosysidp || num_iotasks < 1 || num_iotasks * stride > num_comptasks)
+    if (!iosysidp || num_iotasks < 1 || num_iotasks > num_comptasks ||
+        stride < 1 || stride > num_comptasks ||
```

Comment:
stride > num_comptasks should be ok
e.g. stride = 128, num io tasks = 1, base = 0

Reply:
An existing unit test (in test_iosystem2_simple2.c) has inputs:
stride = 31, num comp tasks = 1, num io tasks = 1, base = 0
It is treated as invalid inputs. If we allow stride > num_comptasks
in this case, the inputs will be valid and that test will fail.

Reply:
The test is probably wrong

Reply:
OK, I will remove that wrong test as well.